File: FestivalJava/src/main/java/Repository/Interfaces/IRepository.java
package Repository.Interfaces;

import java.util.List;

/**
 * Created by Sebi on 09-Mar-17.
 */
public interface IRepository<E, ID> {
    /*
    Saves an entity
     */
    void save(E entity);

    /*
    Deletes an entity
     */
    void delete(ID id);

    /*
    Updates an entity
     */
    void update(ID id, E entity);

    /*
    Gets the entity with the given id
     */
    E getById(ID id);

    /*
    Returns all entities
     */
    List<E> getAll();
}

File: FestivalC#/Festival/Festival/Repository/Interface/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Festival.Repository
{
    public interface IRepository<E, ID>
    {
        void save(E e);
        void delete(ID id);
        E getById(ID id);
        List<E> getAll();
        void update(ID id, E e);
    }
}
